### PR TITLE
Misc cleanups & refactorings, Part 1

### PR DESCRIPTION
**Pull request by Centril**
### modified logic

- `MemberNotFound` and `BadOrigin` in the committee pallet got merged into `NotAMember`.
- `Committee::<T, I>::close` now requires committee membership from the origin.
- Fixed atomicity bugs in `move_portfolio_funds` and the bridge pallet in general.
- `remove_smart_extension` allows itself to move shuffle around SEs for improved performance.

### CODE REVIEW

File: pallets/asset/src/lib.rs [hunk 1]
```diff
@@ -519,7 +519,7 @@ decl_module! {
             // Check total supply here to avoid any later failure
             Self::ensure_create_asset_parameters(&ticker, total_supply)?;
             if !divisible {
-                ensure!(total_supply % ONE_UNIT.into() == 0.into(), Error::<T>::InvalidTotalSupply);
+                ensure!(Self::is_unit_multiple(total_supply), Error::<T>::InvalidTotalSupply);
             }
```
Comment (maxsam4) [inline pallets/asset/src/lib.rs:523]:
```
ensure!(divisible || Self::is_unit_multiple(total_supply), Error::<T>::InvalidTotalSupply);
```

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -1350,6 +1317,25 @@ impl<T: Trait> Module<T> {
         T::MaxNumberOfTMExtensionForAsset::get()
     }
 
+    fn ensure_pia_with_custody(
```
Comment (maxsam4) [inline pallets/asset/src/lib.rs:1320]:
```suggestion
    fn ensure_pia_or_owner_with_custody(
```

**Review by maxsam4 — APPROVED** [review]

File: pallets/asset/src/lib.rs [hunk 3]
```diff
@@ -672,23 +667,14 @@ decl_module! {
...
-            let beneficiary = Self::ensure_pia_or_owner(&ticker, primary_did)?;
-
-            // Ensure that the caller has relevant portfolio permissions
-            let beneficiary_portfolio = PortfolioId::default_portfolio(beneficiary);
-            Portfolio::<T>::ensure_portfolio_custody_and_permission(beneficiary_portfolio, primary_did, secondary_key.as_ref())?;
+                ..
+            } = Self::ensure_pia_or_owner_with_custody(origin, ticker)?;
 
-            Self::_mint(&ticker, sender, beneficiary, value, Some(ProtocolOp::AssetIssue))
+            Self::_mint(&ticker, sender, primary_did, value, Some(ProtocolOp::AssetIssue))
```
[the remark this answers is not in the document]
Comment (adamdossa) [reply pallets/asset/src/lib.rs:677]: In terms of the correct semantics, I think ideally, we should conceptually assume there is always a PIA, just that it defaults to the ticker owner if not explicitly set via a PIA authorisation.

Only the PIA should be able to call `issue` and tokens should always then go to the PIA.

@maxsam4 thoughts?

File: pallets/asset/src/lib.rs [hunk 4]
```diff
@@ -1350,6 +1315,25 @@ impl<T: Trait> Module<T> {
...
+        let skey = data.secondary_key.as_ref();
+
+        // Ensure that the secondary key has asset permission
+        Self::ensure_asset_perms(skey, &ticker)?;
+
+        // Ensure that the sender is the PIA or the token owner and returns the PIA address.
+        let pia = Self::ensure_pia_or_owner(&ticker, data.primary_did)?;
+
+        // Ensure that the caller has relevant portfolio permissions
+        let portfolio = PortfolioId::default_portfolio(pia);
```
Comment (satyamakgec) [inline pallets/asset/src/lib.rs:1332]: I think here it get assumed that PIA and owner will always be the same but it is not always true. here owner did should be used.

File: pallets/asset/src/lib.rs [hunk 5]
```diff
@@ -704,30 +690,14 @@ decl_module! {
...
-            let PermissionedCallOriginData {
-                primary_did: did,
-                secondary_key,
-                ..
-            } = Identity::<T>::ensure_origin_call_permissions(origin)?;
-
-            // Ensure that the secondary key has asset permission
-            Self::ensure_asset_perms(secondary_key.as_ref(), &ticker)?;
+            // Ensure origin is PIA with custody and permissions for default portfolio.
+            let pia = Self::ensure_pia_or_owner_with_custody(origin, ticker)?.primary_did;
```
Comment (satyamakgec) [inline pallets/asset/src/lib.rs:694]: This assumption never stands true always. or we assume that only pia is only allowed to call the redeem function.

File: pallets/asset/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxsam4) [reply pallets/asset/src/lib.rs:677]: Yeah, that makes sense. We'll need the same change in the `redeem` function as well.

[the remark this answers is not in the document]
Comment (adamdossa) [reply pallets/asset/src/lib.rs:677]: Added PR:
https://github.com/PolymathNetwork/Polymesh/pull/834
to address this.

File: pallets/asset/src/lib.rs (hunk 4 above)
Comment (Centril) [reply pallets/asset/src/lib.rs:1332]: It's assumed that `data.primary_did == pia` always holds, which is the case:
```rust
    pub fn ensure_pia_or_owner(
        ticker: &Ticker,
        did: IdentityId,
    ) -> Result<IdentityId, DispatchError> {
        Self::token_details(&ticker)
            .primary_issuance_agent
            .filter(|pia| *pia == did)
            .or_else(|| Self::is_owner(&ticker, did).then(|| did))
            .ok_or_else(|| Error::<T>::Unauthorized.into())
    }
```

File: pallets/asset/src/lib.rs (hunk 5 above)
Comment (Centril) [reply pallets/asset/src/lib.rs:694]: See above.

File: pallets/asset/src/lib.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Centril) [reply pallets/asset/src/lib.rs:677]: > with this change, I think if the owner calls `issue`, they will receive the minted tokens, not the PIA

N.B. the PR does not imply semantic changes here atop of `develop`; rather, the pre-existing semantics in `develop` seem to be different from what we want.

**Review by adamdossa — APPROVED** [review]

**Pull request merged**